var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -47,15 +49,15 @@ class Julia(MakefilePackage):
     # Note, we just use link_llvm_dylib so that we not only get a libLLVM,
     # but also so that llvm-config --libfiles gives only the dylib. Without
     # it it also gives static libraries, and breaks Julia's build.
-    depends_on(
-        "llvm"
-        " targets=amdgpu,bpf,nvptx,webassembly"
-        " version_suffix=jl +link_llvm_dylib ~internal_unwind"
-    )
+    depends_on("llvm targets=amdgpu,bpf,nvptx,webassembly version_suffix=jl +link_llvm_dylib")
     depends_on("libuv", when="@:1.7")
     depends_on("libuv-julia@1.42.0", when="@1.8.0:1.8.1")
     depends_on("libuv-julia@1.44.2", when="@1.8.2:")
 
+    # Do not use internal unwind.  We need to use a conflict, because
+    # `internal_unwind` is defined only when `+clang`.
+    conflicts("llvm+internal_unwind")
```

Comment:
@haampie you suggested me to do this.  It works:
```console
% spack spec julia ^llvm+clang+internal_unwind
==> Error: julia conflicts with 'llvm +internal_unwind'
```
but `spack audit` [isn't happy](https://github.com/spack/spack/actions/runs/4400269515/jobs/7705415753#step:6:13):
```console
% spack audit packages julia
PKG-DIRECTIVES: 1 issue found
1. julia: wrong variant in "conflicts" directive
    the variant 'internal_unwind' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/julia/package.py
```
I thought the problem was that `internal_unwind` is defined only when `+clang`, but
```py
conflicts("llvm+clang+internal_unwind")
```
doesn't work either
```console
% spack audit packages julia
PKG-DIRECTIVES: 2 issues found
1. julia: wrong variant in "conflicts" directive
    the variant 'clang' does not exist
    in /home/mose/repo/spack/var/spack/repos/builtin/packages/julia/package.py
2. julia: wrong variant in "conflicts" directive
    the variant 'internal_unwind' does not exist
    in /home/mose/repo/spack/var/spack/repos/builtin/packages/julia/package.py
```
and the `clang` variant is always defined https://github.com/spack/spack/blob/31de7ea56cb9716f31b33dfc9a5c3db188f727df/var/spack/repos/builtin/packages/llvm/package.py#L78-L80 So now I'm not quite sure how to make `spack audit` happy.

Reply:
Can you check with:
```
conflicts("^llvm+internal_unwind")
```
? If that works I'd not be sure whether to consider this a bug or a feature :slightly_smiling_face: (the `^` denotes a dependency and the conflict is on a dependency, though it should be clear from the fact that the package name is not "julia")

Reply:
That worked, thanks!  But yeah, I think the check is kind of confusing if it's looking for the `internal_unwind` variant in the `julia` package and not `llvm`

Reply:
Is that a regression due to #35681? :thinking: have to check.